packages/cubejs-schema-compiler/src/adapter/ClickHouseQuery.ts

```diff
@@ -168,6 +184,35 @@ export class ClickHouseQuery extends BaseQuery {
     return `${fieldAlias} ${direction}`;
   }
 
+  public override orderBy() {
+    //
+    // ClickHouse orders string by bytes, so we need to use COLLATE 'en' to order by string
+    //
+    if (R.isEmpty(this.order)) {
+      return '';
+    }
+
+    const collation = getEnv('clickhouseSortCollation') || 'en';
+
+    const orderByString = R.pipe(
+      R.map((order) => {
+        let orderString = this.orderHashToString(order);
+        if (this.getFieldType(order) === 'string') {
+          orderString = `${orderString} COLLATE '${collation}'`;
+        }
+        return orderString;
+      }),
+      R.reject(R.isNil),
+      R.join(', ')
+    )(this.order);
```

Comment:
What if someone uses ascii and/or doesn't need any collation sorting at all? Also with this flow, we preserve the backward compatibility because, in other cases, someone might be surprised with the new results if their data is in non-en locale...

```suggestion
    const collation = getEnv('clickhouseSortCollation');

    const orderByString = R.pipe(
      R.map((order) => {
        let orderString = this.orderHashToString(order);
        if (this.getFieldType(order) === 'string') && collation !== '' {
          orderString = `${orderString} COLLATE '${collation}'`;
        }
        return orderString;
      }),
      R.reject(R.isNil),
      R.join(', ')
    )(this.order);
```